### ISSUE TITLE
UX Update on training flow revamp - subset distribution

### ISSUE THREAD

**Issue by jpggvilaca** [open-edge-platform/geti_v2#1019]
The issue is that the subset proportions (train + validation + test) are determined at the task level, which means that once any model is trained, the subset distribution cannot be changed for any future model unless you train from scratch and enable "Reshuffle subsets."
 
In practice, this means that changing the subset distribution

<img width="989" height="441" alt="Image" src="https://github.com/user-attachments/assets/8d3ae21e-9258-4352-83b2-53d8a52f5f6c" />

won't have any effect unless you also select "train from scratch" and "reshuffle subsets."
 <img width="986" height="476" alt="Image" src="https://github.com/user-attachments/assets/1022a9e3-15c6-4f4b-b68b-0fba37477035" />
 
I believe that if we change the backend logic to fit the current UX, we might end up with inconsistencies in the training data, such as mixing the training set with the test set in subsequent rounds of training.
So, I think it's better to adjust the UX so that when the user changes the distribution, "train from scratch" and "reshuffle subsets" are also enabled, and a warning is prompted to the user

Context: https://github.com/open-edge-platform/geti/issues/955
